Spanner/src/Database.php

```diff
@@ -1146,19 +1171,70 @@ public function delete($table, KeySet $keySet, array $options = [])
      *
      * ```
      * // Array parameters which may be null or empty must include the array value type.
+     * use Google\Cloud\Spanner\ArrayType;
+     *
      * $result = $database->execute('SELECT @emptyArrayOfIntegers as numbers', [
      *     'parameters' => [
      *         'emptyArrayOfIntegers' => []
      *     ],
      *     'types' => [
-     *         'emptyArrayOfIntegers' => [Database::TYPE_ARRAY, Database::TYPE_INT64]
+     *         'emptyArrayOfIntegers' => new ArrayType(Database::TYPE_INT64)
      *     ]
      * ]);
      *
      * $row = $result->rows()->current();
      * $emptyArray = $row['numbers'];
      * ```
      *
+     * ```
+     * // Struct parameters must always declare types.
+     * use Google\Cloud\Spanner\StructType;
+     *
+     * $result = $database->execute('SELECT @userStruct.firstName, @userStruct.lastName', [
+     *     'parameters' => [
+     *         'userStruct' => [
+     *             'firstName' => 'John',
+     *             'lastName' => 'Testuser'
+     *         ]
+     *     ],
+     *     'types' => [
+     *         'userStruct' => (new StructType())
+     *             ->add('firstName', Database::TYPE_STRING)
+     *             ->add('lastName', Database::TYPE_STRING)
+     *     ]
+     * ]);
+     *
+     * $row = $result->rows()->current();
+     * $fullName = $row['firstName'] . ' ' . $row['lastName']; // `John Testuser`
+     * ```
+     *
+     * ```
+     * // If a struct contains unnamed fields, or multiple fields with the same
+     * // name, it must be defined using {@see Google\Cloud\Spanner\StructValue}.
+     * use Google\Cloud\Spanner\Result;
+     * use Google\Cloud\Spanner\StructValue;
+     * use Google\Cloud\Spanner\StructType;
+     *
+     * $res = $database->execute('SELECT * FROM UNNEST(ARRAY(SELECT @structParam))', [
+     *     'parameters' => [
+     *         'structParam' => (new StructValue)
+     *             ->add('foo', 'bar')
+     *             ->add('foo', 2)
+     *             ->addUnnamed('this field is unnamed')
```

Comment:
why not just call this add which just takes a single argument.

Reply:
Because PHP does not support method overloading, there is no good way to distinguish between a field with a value of `null` and an unnamed field while still preserving a good method signature which supports code intelligence.

i.e. `add('this field is unnamed')` and `add('this field is unnamed', null)` would not be easily distinguished internally without suffering other potential drawbacks.